### PR TITLE
chore: lint code

### PR DESCRIPTION
**Pull request by faddat**
Closes: #113 

- add pebbledb
- use latest pebble
- added a ton of test rigor and fixed linting

This PR lints tm-db

closes: #113

### CODE REVIEW

**Comment by faddat** [comment]
let's do it.  codecov is dead.

**Comment by faddat** [comment]
Note that there's a clear merge order here:

* #112 adds pebble for the last time.  I cannot help those who don't help themselves:

```log
Test Case       Backend SetTime ReadTime        WriteTime       ConcurrentTime
Test Case: Keys_10000_Values_512
Total time for pebbledb: 0.09 seconds
Total time for goleveldb: 0.11 seconds
Faster Backend: pebbledb
-----------------------------
Test Case: Keys_10000_Values_512
Total time for pebbledb: 0.09 seconds
Total time for goleveldb: 0.11 seconds
Faster Backend: pebbledb
-----------------------------
Test Case: Keys_10000_Values_1024
Total time for pebbledb: 0.10 seconds
Total time for goleveldb: 0.14 seconds
Faster Backend: pebbledb
-----------------------------
Test Case: Keys_10000_Values_1024
Total time for pebbledb: 0.10 seconds
Total time for goleveldb: 0.14 seconds
Faster Backend: pebbledb
-----------------------------
Test Case: Keys_10000_Values_2048
Total time for pebbledb: 0.11 seconds
Total time for goleveldb: 0.23 seconds
Faster Backend: pebbledb
-----------------------------
Test Case: Keys_10000_Values_2048
Total time for pebbledb: 0.11 seconds
Total time for goleveldb: 0.23 seconds
Faster Backend: pebbledb
-----------------------------
Test Case: Keys_10000_Values_4096
Total time for pebbledb: 0.13 seconds
Total time for goleveldb: 0.43 seconds
Faster Backend: pebbledb
-----------------------------
Test Case: Keys_10000_Values_4096
Total time for pebbledb: 0.13 seconds
Total time for goleveldb: 0.43 seconds
Faster Backend: pebbledb
-----------------------------
Test Case: Keys_25000_Values_512
Total time for pebbledb: 0.16 seconds
Total time for goleveldb: 0.29 seconds
Faster Backend: pebbledb
-----------------------------
Test Case: Keys_25000_Values_512
Total time for pebbledb: 0.16 seconds
Total time for goleveldb: 0.29 seconds
Faster Backend: pebbledb
-----------------------------
Test Case: Keys_25000_Values_1024
Total time for pebbledb: 0.18 seconds
Total time for goleveldb: 0.38 seconds
Faster Backend: pebbledb
-----------------------------
Test Case: Keys_25000_Values_1024
Total time for pebbledb: 0.18 seconds
Total time for goleveldb: 0.38 seconds
Faster Backend: pebbledb
-----------------------------
Test Case: Keys_25000_Values_2048
Total time for pebbledb: 0.24 seconds
Total time for goleveldb: 0.66 seconds
Faster Backend: pebbledb
-----------------------------
Test Case: Keys_25000_Values_2048
Total time for pebbledb: 0.24 seconds
Total time for goleveldb: 0.66 seconds
Faster Backend: pebbledb
-----------------------------
Test Case: Keys_25000_Values_4096
Total time for pebbledb: 0.21 seconds
Total time for goleveldb: 1.18 seconds
Faster Backend: pebbledb
-----------------------------
Test Case: Keys_25000_Values_4096
Total time for pebbledb: 0.21 seconds
Total time for goleveldb: 1.18 seconds
Faster Backend: pebbledb
-----------------------------
Test Case: Keys_50000_Values_512
Total time for pebbledb: 0.28 seconds
Total time for goleveldb: 0.56 seconds
Faster Backend: pebbledb
-----------------------------
Test Case: Keys_50000_Values_512
Total time for pebbledb: 0.28 seconds
Total time for goleveldb: 0.56 seconds
Faster Backend: pebbledb
-----------------------------
Test Case: Keys_50000_Values_1024
Total time for pebbledb: 0.32 seconds
Total time for goleveldb: 0.80 seconds
Faster Backend: pebbledb
-----------------------------
Test Case: Keys_50000_Values_1024
Total time for pebbledb: 0.32 seconds
Total time for goleveldb: 0.80 seconds
Faster Backend: pebbledb
-----------------------------
Test Case: Keys_50000_Values_2048
Total time for pebbledb: 0.33 seconds
Total time for goleveldb: 1.35 seconds
Faster Backend: pebbledb
-----------------------------
Test Case: Keys_50000_Values_2048
Total time for pebbledb: 0.33 seconds
Total time for goleveldb: 1.35 seconds
Faster Backend: pebbledb
-----------------------------
Test Case: Keys_50000_Values_4096
Total time for pebbledb: 0.35 seconds
Total time for goleveldb: 3.05 seconds
Faster Backend: pebbledb
-----------------------------
Test Case: Keys_50000_Values_4096
Total time for pebbledb: 0.35 seconds
Total time for goleveldb: 3.05 seconds
Faster Backend: pebbledb
-----------------------------
Test Case: Keys_75000_Values_512
Total time for pebbledb: 0.41 seconds
Total time for goleveldb: 2.04 seconds
Faster Backend: pebbledb
-----------------------------
Test Case: Keys_75000_Values_512
Total time for pebbledb: 0.41 seconds
Total time for goleveldb: 2.04 seconds
Faster Backend: pebbledb
-----------------------------
Test Case: Keys_75000_Values_1024
Total time for pebbledb: 0.48 seconds
Total time for goleveldb: 1.56 seconds
Faster Backend: pebbledb
-----------------------------
Test Case: Keys_75000_Values_1024
Total time for pebbledb: 0.48 seconds
Total time for goleveldb: 1.56 seconds
Faster Backend: pebbledb
-----------------------------
Test Case: Keys_75000_Values_2048
Total time for pebbledb: 0.45 seconds
Total time for goleveldb: 2.32 seconds
Faster Backend: pebbledb
-----------------------------
Test Case: Keys_75000_Values_2048
Total time for pebbledb: 0.45 seconds
Total time for goleveldb: 2.32 seconds
Faster Backend: pebbledb
-----------------------------
Test Case: Keys_75000_Values_4096
Total time for pebbledb: 0.62 seconds
Total time for goleveldb: 7.56 seconds
Faster Backend: pebbledb
-----------------------------
Test Case: Keys_75000_Values_4096
Total time for pebbledb: 0.62 seconds
Total time for goleveldb: 7.56 seconds
Faster Backend: pebbledb
-----------------------------
Test Case: Keys_100000_Values_512
Total time for pebbledb: 0.50 seconds
Total time for goleveldb: 1.17 seconds
Faster Backend: pebbledb
-----------------------------
Test Case: Keys_100000_Values_512
Total time for pebbledb: 0.50 seconds
Total time for goleveldb: 1.17 seconds
Faster Backend: pebbledb
-----------------------------
Test Case: Keys_100000_Values_1024
Total time for pebbledb: 0.55 seconds
Total time for goleveldb: 1.74 seconds
Faster Backend: pebbledb
-----------------------------
Test Case: Keys_100000_Values_1024
Total time for pebbledb: 0.55 seconds
Total time for goleveldb: 1.74 seconds
Faster Backend: pebbledb
-----------------------------
Test Case: Keys_100000_Values_2048
Total time for pebbledb: 0.53 seconds
Total time for goleveldb: 3.34 seconds
Faster Backend: pebbledb
-----------------------------
Test Case: Keys_100000_Values_2048
Total time for pebbledb: 0.53 seconds
Total time for goleveldb: 3.34 seconds
Faster Backend: pebbledb
-----------------------------
Test Case: Keys_100000_Values_4096
Total time for pebbledb: 0.66 seconds
Total time for goleveldb: 7.25 seconds
Faster Backend: pebbledb
-----------------------------
Test Case: Keys_100000_Values_4096
Total time for pebbledb: 0.66 seconds
Total time for goleveldb: 7.25 seconds
Faster Backend: pebbledb
-----------------------------
PASS
```

Please let me know if those are real-world-ish.  I think they are.  Mainly pebble is doing 4-10x.

------------


* #114 

it lints the repository because it is easier to work on well formatted, linted code


-------------


* #115 

goes about the difficult task of writing comparative benchmarks

**Comment by melekes** [comment]
Is it possible to exclude Pebble from this PR and make it only about lint changes?

**Comment by melekes** [comment]
https://github.com/cometbft/cometbft-db/pull/112 needs to be merged first.

**Comment by faddat** [comment]
@melekes yes, I agree.

File: .changelog/unreleased/feat/112.md [hunk 1]
```diff
@@ -0,0 +1 @@
+Add Support for github.com/cockroachdb/pebble
```
Comment (melekes) [inline .changelog/unreleased/feat/112.md:1]: please remove

File: .golangci.yml [hunk 2]
```diff
@@ -1,51 +1,113 @@
...
+        - staticcheck
+      text: "SA1019:" # silence errors on usage of deprecated funcs
+
+  max-issues-per-linter: 10000
+  max-same-issues: 10000
+
 linters-settings:
-  errcheck:
-    check-blank: true
-  depguard:
```
Comment (melekes) [inline .golangci.yml:33]: do you mind bringing this back?

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,11 @@
 # CHANGELOG
 
+## [Unreleased]
```
Comment (melekes) [inline CHANGELOG.md:3]: please remove

File: .golangci.yml (hunk 2 above)
Comment (faddat) [reply .golangci.yml:33]: Few minutes.

**Comment by faddat** [comment]
Oh that's okay I'll fix it

File: .golangci.yml (hunk 2 above)
Comment (faddat) [reply .golangci.yml:33]: generally, depguard isn't in my lint religion -- did you want it back & configured?

**Comment by faddat** [comment]
I'll add a make test-pebble to default test runs..

**Review by melekes — APPROVED** [review]

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,9 @@
 # CHANGELOG
 
+*Jan 24, 2024*
```
Comment (melekes) [inline CHANGELOG.md:3]: please remove

**Review by melekes — APPROVED** [review]

**Pull request merged**